src/subscription_manager/utils.py

```diff
@@ -62,6 +62,77 @@ def __repr__(self):
         return pprint.pformat(self.as_dict())
 
 
+def three_way_merge(local=None, base=None, remote=None, on_conflict="remote"):
+    """
+    Performs a three-way merge on the local and remote dictionaries with a given base.
+    :param local: The dictionary of the current local values
+    :param base: The dictionary with the values we've last seen
+    :param remote: The dictionary with "their" values
+    :param on_conflict: Either "remote" or "local" or None. If "remote", the remote changes
+                               will win any conflict. If "local", the local changes will win any
+                               conflict. If anything else, an error will be thrown.
+
+    :return: The dictionary of values as merged between the three provided dictionaries.
+    """
+    # Ensure that we at least have a value for each keyword arg. Left each input param as a keyword
```

Comment:
Comment misplaced, FYI, also can use the function args as keyword arg, even without providing defaults in the signature.

i.e. can do

```python
def three_way_merge(local, base, remote, on_conflict="remote")
```